### PR TITLE
Extend provider management feature to cover editing

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Extend provider management feature to cover editing

#### Ticket

relates to [CFP-159 nested error linking](https://dsdmoj.atlassian.net/browse/CFP-159)

#### Why
Happy path coverage for basic functionality to prevent
regressions.

### CODE REVIEW

File: features/provider_management.feature [hunk 1]
```diff
@@ -71,21 +71,39 @@ Feature: Case worker can manage providers
...
+
+    When I click the link 'Edit'
+    Then I should be on the provider manager user edit page
+    And the page should be accessible
+
+    When I fill in 'First name' with 'Bob'
+    And I fill in 'Email' with 'bob.mann@example.com'
+    And I fill in 'Email confirmation' with 'bob.mann@example.com'
+    And I click the button 'Update user'
+    And I save and open screenshot
```
Comment (jsugarman) [inline features/provider_management.feature]: remove

File: features/page_objects/external_user_manage_users_page.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
-class ManageUsersPage < BasePage
+class ExternalUserManageUsersPage < BasePage
```
Comment (jrmhaig) [inline features/page_objects/external_user_manage_users_page.rb:1]: Happy for this to change but I do not see how it is related to the PR. Likewise for the changes to `features/step_definitions/manage_users_steps.rb` and the change to line 10 of `features/support/hooks.rb`. Perhaps it should be split into a separate commit.

**Review by jrmhaig — APPROVED** [review]
